test/unit/org/apache/cassandra/locator/DynamicEndpointSnitchTest.java

```diff
@@ -60,57 +130,229 @@ private static EndpointsForRange full(InetAddressAndPort... endpoints)
+        assertEquals(2, measurements.values().stream().filter(m -> m.probeTimerMillis > 0).count());
+        assertEquals(0, measurements.values().stream().filter(m -> m.probeFuture != null).count());
+
+        // Both requested but non measured hosts should have single interval timers set
+        assertEquals(dsnitch.dynamicLatencyProbeInterval, measurements.get(hosts[3]).probeTimerMillis);
+        assertEquals(dsnitch.dynamicLatencyProbeInterval, measurements.get(hosts[4]).probeTimerMillis);
+
+        dsnitch.schedulePings(measurements);
+        // We should have two scheduled futures now
+        assertEquals(2, measurements.values().stream().filter(m -> m.probeFuture != null).count());
+
+        int numProbes = 0;
+        for (int i = 1; (1 << i) * dsnitch.dynamicLatencyProbeInterval < DynamicEndpointSnitch.MAX_PROBE_INTERVAL_MS; i++)
+        {
+            DynamicEndpointSnitch.AnnotatedMeasurement m3 = measurements.get(hosts[3]);
+            DynamicEndpointSnitch.AnnotatedMeasurement m4 = measurements.get(hosts[4]);
+            // Emulate the probe actually running
+            m3.probeFuture.cancel(false);
+            m4.probeFuture.cancel(false);
+            DynamicEndpointSnitch.calculateProbes(measurements, dsnitch.dynamicLatencyProbeInterval);
+            dsnitch.schedulePings(measurements);
+            assertEquals((1 << i) * dsnitch.dynamicLatencyProbeInterval, m3.probeTimerMillis);
+            assertEquals((1 << i) * dsnitch.dynamicLatencyProbeInterval, m4.probeTimerMillis);
+            numProbes++;
+        }
+
+        // If we did exponential backoff right ... we should have the right number of probes
+        assertTrue(numProbes > 0);
+        assertEquals(numProbes,
+                     Math.floor(Math.log(DynamicEndpointSnitch.MAX_PROBE_INTERVAL_MS / dsnitch.dynamicLatencyProbeInterval) / Math.log(2.0)),
+                     0.01);
+
+        measurements.get(hosts[3]).probeFuture.cancel(false);
+        measurements.get(hosts[4]).probeFuture.cancel(false);
+
+        // Now we should do ten minute probes until we exceed MAX_PROBE_INTERVAL_MS since we last requested ranking
+        int interval = dsnitch.dynamicLatencyProbeInterval;
+        for (int i = (probesBefore + numProbes) * interval; i < DynamicEndpointSnitch.MAX_PROBE_INTERVAL_MS; i += interval)
+        {
+            DynamicEndpointSnitch.calculateProbes(measurements, dsnitch.dynamicLatencyProbeInterval);
+            assertEquals(DynamicEndpointSnitch.MAX_PROBE_INTERVAL_MS, measurements.get(hosts[3]).probeTimerMillis);
+            assertEquals(DynamicEndpointSnitch.MAX_PROBE_INTERVAL_MS, measurements.get(hosts[4]).probeTimerMillis);
+            dsnitch.schedulePings(measurements);
+            measurements.get(hosts[3]).probeFuture.cancel(false);
+            measurements.get(hosts[4]).probeFuture.cancel(false);
+        }
+
+        // Since we haven't measured, we should stop probing
```

Comment:
Haven't requested right? It's true we haven't measured it, but it's not having measured it that makes not probed.

Reply:
Correct, typo on my part. Fixed.